src/app/campaign/store/reducers/campaign.reducer.ts

```diff
@@ -39,17 +38,19 @@ export function reducer(state = initialState, action: CampaignActions): Campaign
         valid
       };
     }
-    case ActionTypes.CAMPAIGN_SET_ADVERTISER: {
-      const { set_advertiser_uri } = action.payload;
+    case ActionTypes.CAMPAIGN_ADD_ADVERTISER_SUCCESS: {
+      const { set_advertiser_uri } = docToAdvertiser(action.payload.doc);
       return {
         ...state,
         localCampaign: {
           ...state.localCampaign,
           set_advertiser_uri
         },
-        changed: true
+        changed: true,
+        valid: true
```

Comment:
Hmm... can you really assume the form is valid here?  There could be other missing+required fields.

Reply:
valid concern! it should just keep valid from existing state. will fix.